HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+* association with a token or the presence of available automatics associations.
+* Airdrops that result in transfers will see the transfers populated in the record token_transfer_list.
+* Airdrops that consume an association slot will be represented in the record automatioc_association field.
+* Airdrops that result in pending transfers will see the intended transfer captures in the record pending_airdrops field.
+*/ 
+message TokenAirdropTransactionBody {
+    /**
+    * The desired list of tokens to airdrop. If any custom fees are assessed,
+    * the ledger will try to deduct them from the payer of this airdrop,
+    * resolving the transaction to INSUFFICIENT_PAYER_BALANCE_FOR_CUSTOM_FEE if
+    * this is not possible.
+    * The sender must have all the succifient balance at the time of the transaction to support this transaction.
+    *
+    * If the receiver account has already associated with the given tokens, then
+    * the fee to transfer will be identical to a token transfer. If the receiver
+    * has open auto-association slots but has not yet associated with the token,
+    * then the payer will pay for the transfer, association, one renewal credit,
+    * and airdrop fee. If the receiver has not associated with the token and has
+    * no open association slot, then the transfer will not deliver the token,
+    * but will create a `pending airdrop` instead, with the receiver account
+    * listed on the `pending airdrop`. The payer will pay for the transfer,
+    * association, one renewal credit, and airdrop fee.
+    * All token_transfers must succeed (result in transfer or pending state) for the transaction to succeed.
+    *
+    * A max list of up to 10 tokens maybe be proposed for airdrop.
+    */
+    repeated TokenTransferList token_transfers = 1;
+}
+```
+
+TokenAirdrop transaction will support the transfers of value in a similar manner to CryptoTransfer with the difference of only supporting tokens (no HBAR transfer support) and supporting the resulting pending transfer state where a CryptoTransfer would have failed due to the lack of association options.
+
+However, it is important to have a separate HAPI transaction type for a few reasons
+
+- the ability to call out the potential separate behaviours based on maxAutomaticAssociation, whereby transfer can end up in a pending transfer state or a consumption of an association slot.
+- cleaner conceptual understanding
+- naming allows the grouping of functionality with Cancel and Claim which complete the pending state a TokenAirdrop may creates
+- explicit specification in record stream.
+- improved developer experience
+
+Similar restrictions around transfer list and aggregation across sender and receiver from CryptoTransfer will be applied to TokenAirdop e.g. max 10 transfers per transaction
+
+At the time of this HIP `TokenAirdropTransaction` will not be supported by the Scheduled Transaction Service. A future HIP may explore this support.
+
+#### Cancel
```

Comment:
The HIP section seems incorrect. Below we have an example of TokenPendingAirdrop, not TokenCancelAidrop